.github/workflows/Unit-Test.yml

```diff
@@ -14,30 +14,57 @@ jobs:
+            if [[ -n ${GLIBC_237_COMPATIBLE_PACKAGES} ]]; then
+              echo "GLIBC_237_COMPATIBLE_PACKAGES=${GLIBC_237_COMPATIBLE_PACKAGES}" >> $GITHUB_ENV
+              echo "PR #${{ github.event.pull_request.number }} has these possibly Glibc 2.37 compatible packages: ${GLIBC_237_COMPATIBLE_PACKAGES}"
+            fi
 
-            # If a package doesnt have a min_glibc value, or if its below 2.37, add it to GLIBC_237_COMPATIBLE_PACKAGES
-            echo "GLIBC_237_COMPATIBLE_PACKAGES=$(for i in ${CHANGED_PACKAGES} ; do grep min_glibc packages/$i.rb | tr -d \' | awk '{exit $2 <= 2.37}' || echo $i ; done | xargs)" >> $GITHUB_ENV
-            echo "PR #${{ github.event.pull_request.number }} has these possibly Glibc 2.37 compatible packages: ${GLIBC_237_COMPATIBLE_PACKAGES}"
+            # If a package has a compatibility of 'all' or one that includes 'x86_64', add it to x86_64_PACKAGES.
+            export x86_64_PACKAGES="$(for i in ${CHANGED_PACKAGES}; do grep -q "compatibility.*all\|x86_64" packages/${i}.rb && echo ${i}; done | xargs)"
+            if [[ -n ${x86_64_PACKAGES} ]]; then
+              echo "x86_64_PACKAGES=${x86_64_PACKAGES}" >> $GITHUB_ENV
+              echo "PR #${{ github.event.pull_request.number }} has these x86_64 compatible packages: ${x86_64_PACKAGES}"
+            fi
 
-            # If a package has a compatibility of 'all' or one that includes 'x86_64', add it to the list.
-            echo "PR #${{ github.event.pull_request.number }} has these x86_64 compatible packages: $(for p in ${CHANGED_PACKAGES}; do grep -q "compatibility 'all'\|compatibility '.*x86_64.*'" packages/${p}.rb && echo ${p}; done | xargs)"
+            ## If a package has a compatibility of 'all' or one that includes 'armv7l', add it to ARMV7L_PACKAGES.
+            export ARMV7L_PACKAGES="$(for i in ${CHANGED_PACKAGES}; do grep -q "compatibility.*all\|armv7l" packages/${i}.rb && echo ${i}; done | xargs)"
+            if [[ -n ${ARMV7L_PACKAGES} ]]; then
+              echo "ARMV7L_PACKAGES=${ARMV7L_PACKAGES}" >> $GITHUB_ENV
+              echo "PR #${{ github.event.pull_request.number }} has these armv7l compatible packages: ${ARMV7L_PACKAGES}"
+            fi
 
-            # If a package has a compatibility of 'all' or one that includes 'armv7l', add it to the list.
-            echo "PR #${{ github.event.pull_request.number }} has these armv7l compatible packages: $(for p in ${CHANGED_PACKAGES}; do grep -q "compatibility 'all'\|compatibility '.*armv7l.*'" packages/${p}.rb && echo ${p}; done | xargs)"
+            ## If a package has a compatibility of 'all' or one that includes 'i686', add it to i686_PACKAGES.
+            export i686_PACKAGES="$(for i in ${CHANGED_PACKAGES}; do grep -q "compatibility.*all\|i686" packages/${i}.rb && echo ${i}; done | xargs)"
```

Comment:
this grep logic is broken, as i mentioned in the previous pr:
![image](https://github.com/user-attachments/assets/53fd2478-753f-4475-b33f-f20435a34b12)

Reply:
Yeah agreed... Still futzing with it...

Reply:
(That's why it is in draft...)

Reply:
Good catch!